### ISSUE TITLE
Bogus "is not a branch target"

### ISSUE THREAD

**Issue by klausler** [flang-compiler/f18#515]
```
1 if(0) 2,2,2
  if(0) 1,2,2
2 continue
end
```
compiled with semantics enabled elicits a bogus complaint `'1' is not a branch target`.

**Comment by jeanPerier** [comment]
This seems to be a deliberate warning currently in f18 originating from:
https://github.com/flang-compiler/f18/blob/bc3c549d48a4166c63fbfb312adf0fc7079adeb1/lib/semantics/resolve-labels.cc#L138

From what I can read in the latest standard containing all these features (f90 ISO/IEC 1539 : 1991), all the restrictions in `IsLegalBranchTarget(const parser::Statement<parser::ActionStmt>)` are actually more related to the depreciated/deleted aspect of these `action-stmt` rather than specific rules that would have made branching to these `action-stmt` depreciated.

Now that `-Mstandard` raise "deprecated usage" warnings for all these `action-stmt`, I would advocate to remove this legacy aspect regarding using them as target branches, this seems redundant.

**Comment by jeanPerier** [comment]
Closed by #593: No warning emitted on the program mentioned in the issue.
